### ISSUE TITLE
not all paths return value [8:01am]

### ISSUE THREAD

**Issue by alloursjournall** [frankcollins3/fcc-mcsft-cSharp#3]
**attempting to do:**
create string returning callback functions that take a randomly generated value and return conditionally rendered messages
("Your subscription is ending soon! Renew Now! || Your subscription is ending in {daysUntilExpired}. Renew @ 10% discount"}

**error:**
fixed this already. didn't know. they had to be static strings as well.
error CS0236: A field initializer cannot reference the non-static field, method, or property 'Program.random' [/Users/me/Desktop/CsharpProjects/TestProject/TestProject.csproj]

**proposed approach:**
possibly since the else block hasn't been made. keeping issue open since issues keep popping up.

**Comment by alloursjournall** [comment]
👍 
![Screen Shot 2023-09-06 at 8 02 09 AM](https://github.com/frankcollins3/fcc-mcsft-c-/assets/73137934/d54ff025-751c-49aa-bb38-17d2a3c5a692)

compiler seems like it's being silly because every possible case has been covered 
![Screen Shot 2023-09-06 at 8 05 01 AM](https://github.com/frankcollins3/fcc-mcsft-c-/assets/73137934/3ae7cf84-7b3e-4a63-9d22-cf35e2832b57)

// seems that by convention else must be provided if the current block of if code is else if.
// even if all possibly cases have been provided return options in a non static function.

[8:06am]

**Comment by alloursjournall** [comment]
was hung up because 2 && 3 definitely looked like correct answers.
I ended up picking option 3 and failed that question but this very issue was solved by:
providing else value in event that else if doesn't have one.

The issue that is returned is "Not all paths return value"
![Screen Shot 2023-09-06 at 8 12 21 AM](https://github.com/frankcollins3/fcc-mcsft-c-/assets/73137934/e8a2a765-7d3c-4ff2-8054-2c1882b6b21b)

probably missing the edge case a number > 5 or so but moving on.
[8:14am]
